sdk/internal/perf/testdata/perf/sleep_perf.go

```diff
@@ -58,8 +58,13 @@ func (g *globalSleepPerfTest) NewPerfTest(ctx context.Context, options *perf.Per
 }
 
 func (s *sleepPerfTest) Run(ctx context.Context) error {
-	time.Sleep(s.sleepInterval)
-	s.sleepInterval = time.Duration(float64(s.sleepInterval.Nanoseconds())*sleepTestOpts.iterationGrowthFactor) * time.Nanosecond
+	select {
+	case <-ctx.Done():
+		return nil
+	default:
+		time.Sleep(s.sleepInterval)
+		s.sleepInterval = time.Duration(float64(s.sleepInterval.Nanoseconds())*sleepTestOpts.iterationGrowthFactor) * time.Nanosecond
```

Comment:
This case is very gnarly. I mentioned it above, but if there was a way to just keep it as a time.Duration value it'd probably simplify some things. (perhaps iterationGrowthFactor expects floats though so that'd make it difficult)?

Reply:
iterationGrowthFactor is a `float64` which makes this difficult. I'm going to leave it as is for now.